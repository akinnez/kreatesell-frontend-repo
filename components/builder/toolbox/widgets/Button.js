import React from 'react'
import { useNode } from '@craftjs/core'
import styles from './Index.module.scss'



const Button = ({text = "Button",fontSize,color,fontFamily,fontWeight})=>{

    const {connectors:{connect,drag},actions:{setProp}} = useNode()

    return(
      <>
        <div ref={ref=>connect(drag(ref))}>
           <button className={styles.button}><span>{text}</span></button>          
        </div>
        
        </>
    )
}

export default Button