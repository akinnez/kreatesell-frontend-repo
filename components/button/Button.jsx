import styles from "./Button.module.scss";

export const Button = ({
	type = "submit",
	text = "Submit",
	loading,
	disabled,
	bgColor = "white",
	icon,
	leftIcon,
	...rest
}) => {
	return (
		<button
			{...rest}
			disabled={loading || disabled}
			type={type}
			className={`
			${rest.className}
			${styles.button}
			${bgColor === "white" && styles.btnWhite} 
			${bgColor === "blue" && styles.btnBlue}
			${bgColor === "primaryBlue" && styles.primaryBlue}
			`}
		>
			<div className={styles.buttonValue}>
				{leftIcon && <div className={styles.buttonIcon}>{leftIcon}</div>}
				{text}
				{icon && <div className={styles.buttonIcon}>{icon}</div>}
			</div>
		</button>
	);
};
