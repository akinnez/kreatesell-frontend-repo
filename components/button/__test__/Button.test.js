import renderer from "react-test-renderer";
import { Button } from "../Button";

describe("Button based on background-color passed", () => {
  it("renders a correct snapshot of <Button  /> with no Icons and defaults of text, bgColor and type props used in the render", () => {
    const btn = renderer.create(<Button />).toJSON();
    expect(btn).toMatchSnapshot();
  });
  it("renders a correct snapshot of <Button  /> with bgColor='blue' and  defaults of text ,type props used in the render", () => {
    const btn = renderer.create(<Button bgColor="blue" />).toJSON();
    expect(btn).toMatchSnapshot();
  });
  it("renders a correct snapshot of <Button  /> with bgColor='primaryBlue' and  defaults of text ,type props used in the render", () => {
    const btn = renderer.create(<Button bgColor="primaryBlue" />).toJSON();
    expect(btn).toMatchSnapshot();
  });
});

describe("Button snapshot when passed icons and a backgroundColor", () => {
  it("renders a snapshot of <Button /> when passed in only the leftIcon prop and other default props", () => {
    const btn = renderer.create(<Button leftIcon={true} />).toJSON();
    expect(btn).toMatchSnapshot();
  });
  it("renders a snapshot of <Button /> when passed in only the icon prop and other default props", () => {
    const btn = renderer.create(<Button icon={true} />).toJSON();
    expect(btn).toMatchSnapshot();
  });
});

describe("Button", () => {
  it("renders a snapshot of Button when passed both icon props and default props", () => {
    const btn = renderer
      .create(<Button icon={true} leftIcon={true} />)
      .toJSON();
    expect(btn).toMatchSnapshot();
  });
});
