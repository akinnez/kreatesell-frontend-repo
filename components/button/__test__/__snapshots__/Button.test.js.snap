// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button based on background-color passed renders a correct snapshot of <Button  /> with bgColor='blue' and  defaults of text ,type props used in the render 1`] = `
<button
  className="
			undefined
			button
			false 
			btnBlue
			false
			"
  type="submit"
>
  <div
    className="buttonValue"
  >
    Submit
  </div>
</button>
`;

exports[`Button based on background-color passed renders a correct snapshot of <Button  /> with bgColor='primaryBlue' and  defaults of text ,type props used in the render 1`] = `
<button
  className="
			undefined
			button
			false 
			false
			primaryBlue
			"
  type="submit"
>
  <div
    className="buttonValue"
  >
    Submit
  </div>
</button>
`;

exports[`Button based on background-color passed renders a correct snapshot of <Button  /> with no Icons and defaults of text, bgColor and type props used in the render 1`] = `
<button
  className="
			undefined
			button
			btnWhite 
			false
			false
			"
  type="submit"
>
  <div
    className="buttonValue"
  >
    Submit
  </div>
</button>
`;

exports[`Button renders a snapshot of Button when passed both icon props and default props 1`] = `
<button
  className="
			undefined
			button
			btnWhite 
			false
			false
			"
  type="submit"
>
  <div
    className="buttonValue"
  >
    <div
      className="buttonIcon"
    />
    Submit
    <div
      className="buttonIcon"
    />
  </div>
</button>
`;

exports[`Button snapshot when passed icons and a backgroundColor renders a snapshot of <Button /> when passed in only the icon prop and other default props 1`] = `
<button
  className="
			undefined
			button
			btnWhite 
			false
			false
			"
  type="submit"
>
  <div
    className="buttonValue"
  >
    Submit
    <div
      className="buttonIcon"
    />
  </div>
</button>
`;

exports[`Button snapshot when passed icons and a backgroundColor renders a snapshot of <Button /> when passed in only the leftIcon prop and other default props 1`] = `
<button
  className="
			undefined
			button
			btnWhite 
			false
			false
			"
  type="submit"
>
  <div
    className="buttonValue"
  >
    <div
      className="buttonIcon"
    />
    Submit
  </div>
</button>
`;
